tapdb/multiverse.go

```diff
@@ -356,6 +422,81 @@ func (b *MultiverseStore) RootNode(ctx context.Context,
 	return rootNode, nil
 }
 
+// UniverseRootNode returns the Universe root node for the given asset ID.
+func (b *MultiverseStore) UniverseRootNode(ctx context.Context,
+	id universe.Identifier) (universe.Root, error) {
+
+	// First, we'll check the root node cache to see if we already have it.
+	rootNode := b.rootNodeCache.fetchRoot(id)
+	if rootNode != nil {
+		return *rootNode, nil
+	}
+
+	var universeRoot UniverseRoot
+
+	universeNamespace := id.String()
+
+	readTx := NewBaseUniverseReadTx()
+
+	dbErr := b.db.ExecTx(ctx, &readTx, func(db BaseMultiverseStore) error {
+		dbRoot, err := db.FetchUniverseRoot(ctx, universeNamespace)
+		if err != nil {
+			return err
+		}
+
+		universeRoot = dbRoot
+		return nil
+	})
+	switch {
+	case errors.Is(dbErr, sql.ErrNoRows):
+		return universe.Root{}, universe.ErrNoUniverseRoot
+	case dbErr != nil:
+		return universe.Root{}, dbErr
+	}
+
+	var nodeHash mssmt.NodeHash
+	copy(nodeHash[:], universeRoot.RootHash[:])
+	smtNode := mssmt.NewComputedNode(
+		nodeHash, uint64(universeRoot.RootSum),
+	)
+
+	return universe.Root{
```

Comment:
Since we fetched the root node from the DB, do we also want to cache it now? Or is that done on another layer?